Bottom: 8592353aa73e9e23381ffa345c6a2f3baecec963
Top:    7e5877605deb58980ac0a92458f9aa2e0b3d45e5
Author: Ian Kent <raven@themaw.net>
Date:   2013-06-08 09:17:56 +0800

Refresh of autofs-5.0.7-fix-potential-null-dereference-in-lookup_mount.patch

---

diff --git a/modules/lookup_sss.c b/modules/lookup_sss.c
index 013d3b4..003ec1f 100644
--- a/modules/lookup_sss.c
+++ b/modules/lookup_sss.c
@@ -673,7 +673,7 @@ int lookup_mount(struct autofs_point *ap, const char *name, int name_len, void *
 				       mapent, ctxt->parse->context);
 	if (ret) {
 		time_t now = time(NULL);
-		int rv = CHE_FAIL;
+		int rv = CHE_OK;
 
 		/* Record the the mount fail in the cache */
 		cache_writelock(mc);
@@ -682,7 +682,8 @@ int lookup_mount(struct autofs_point *ap, const char *name, int name_len, void *
 			rv = cache_update(mc, source, key, NULL, now);
 		if (rv != CHE_FAIL) {
 			me = cache_lookup_distinct(mc, key);
-			me->status = now + ap->negative_timeout;
+			if (me)
+				me->status = now + ap->negative_timeout;
 		}
 		cache_unlock(mc);
 		return NSS_STATUS_TRYAGAIN;
